docs/Userguide_sharing_data.rst

```diff
@@ -0,0 +1,101 @@
+Sharing Data with ACLs
+======================
+
+As an illustrative example, to allow ``$USER`` to share with ``$USER2`` in
+``rwx`` a hierarchy ``/network/scratch/$USER/X/Y/Z/...`` with ACLs:
+
+----
+
+| Grant **oneself** permissions to access any **future** files/folders created
+  by the other *(or oneself)*
```

Comment:
The `$USER` / `$USER2` paradigm in the intro is good to understand what's up, this `oneself` / `other` / `another` version is less clear IMHO

Reply:
Also, I couldn't understand this sentence before the 4th read.

Is this saying: "how to make sure _I_ (`$USER`), will be able to `rwx` the files created by someone else (`$USER2`) if they put anything in `/network/scratch/$USER/X/Y/Z/`"

If that's the case, I think the first item in this tutorial should be how to _allow_ `$USER2` to even have access to `/network/scratch/$USER/X/Y/Z/`

Reply:
@vict0rsch There is a careful reasoning I made for executing those commands in that specific order, it's not by accident.

The idea behind granting *yourself* access to all future files created by (you or other) *first* is that if you don't do that, then in the gap between your execution of that command and the next one, that other person could race you to create files you mysteriously can't delete.

Granting rights to even access the shared workspace to the other person *last* ensures that all ACLs have been readied beforehand and no race condition is possible at the moment of the "unlock".

Reply:
So this is for security reasons? It's interesting. Put this^ as a `..note:` no? A good doc is also a good place to learn 😛 But I understand now thanks

Reply:
It's not for security reasons but for safety reasons. Not sure we should add too much explanations about the examples.

Reply:
@btravouillon In the end I did supply such an explanation, see [here](https://mila-docs-staging--121.org.readthedocs.build/en/121/Userguide.html#sharing-data-with-acls)

I think the current notes strike the right balance between explaining why this should probably not be skipped and delving too deep in the weeds.

Reply:
Well, the comment was not resolved. I just noticed while doing the (hopefully) last review.